source/Cosmos.HAL2/BlockDevice/AtaPio.cs

```diff
@@ -250,20 +250,22 @@ public Status SendCmd(Cmd aCmd, bool aThrowOnError)
 			return xStatus;
 		}
 
-		protected string GetString(UInt16[] aBuffer, int aIndexStart, int aStringLength)
+		protected string GetString(ushort[] aBuffer, int aIndexStart, int aStringLength)
 		{
-			// Would be nice to convert to byte[] and use
-			// new string(ASCIIEncoding.ASCII.GetChars(xBytes));
-			// But it requires some code Cosmos doesnt support yet
-			var xChars = new char[aStringLength];
-			for (int i = 0; i < aStringLength / 2; i++)
-			{
-				UInt16 xChar = aBuffer[aIndexStart + i];
-				xChars[i * 2] = (char)(xChar >> 8);
-				xChars[i * 2 + 1] = (char)xChar;
-			}
-			return new string(xChars);
-		}
+            //Convert ushort[] to byte[]
+            var array = new List<byte>();
+            for (int i = aIndexStart; i < aIndexStart + (aStringLength/2); i++)
+            {
+                var item = aBuffer[i];
+                var bytes = BitConverter.GetBytes(item);
+                for (int i2 = 1; i2 >= 0; i2--)
```

Comment:
I think the intent is clearer if you just do `array.Add(bytes[1]); array.Add(bytes[0]);` than use a for loop.

Reply:
I don't even know why I did not even think about that.